### PR TITLE
fix logout problem with multipe flightpath docking widget

### LINKED ISSUE #2309: multipe flightpath docking widget on logout traceback
when you have worked with the docking widget then closed topview and did logout.

click some operations, click a local flightpath, click again on a operation then logout

```
Fatal error in MSS 8.3.4 on Linux-6.5.0-10027-tuxedo-x86_64-with-glibc2.35
Python 3.11.6 | packaged by conda-forge | (main, Oct  3 2023, 10:40:35) [GCC 12.3.0]

Please report bugs in MSS to https://github.com/Open-MSS/MSS

Information about the fatal error:

Traceback (most recent call last):
  File "/home/user/PycharmProjects/MSS/mslib/msui/topview.py", line 365, in <lambda>
    self.mainwindow_signal_logout_mscolab.connect(lambda: self.signal_logout_mscolab.emit())
                                                          ^^^^^^^^^^^^^^^^^^^^^^^^^^
RuntimeError: wrapped C/C++ object of type MSUITopViewWindow has been deleted
```

close_topview_then_logout

### PR DESCRIPTION
**Pull request by ReimarBauer**
**Purpose of PR?**:

Fixes #2309

**Does this PR introduce a breaking change?**

**If the changes in this PR are manually verified, list down the scenarios covered:**:

**Additional information for reviewer?** :
_Mention if this PR is part of any design or a continuation of previous PRs_

**Does this PR results in some Documentation changes?**
_If yes, include the list of Documentation changes_

**Checklist:**
- [X] Bug fix. Fixes #2309
- [ ] New feature (Non-API breaking changes that adds functionality)
- [ ] PR Title follows the convention of  `<type>: <subject>`
- [ ] Commit has unit tests

### CODE REVIEW

File: mslib/msui/topview.py [hunk 1]
```diff
@@ -201,9 +201,6 @@ def __init__(self, parent=None, mainwindow=None, model=None, _id=None,
         self.mainwindow_signal_permission_revoked = mainwindow.signal_permission_revoked
         self.mainwindow_signal_render_new_permission = mainwindow.signal_render_new_permission
         self.mainwindow_signal_activate_flighttrack = mainwindow.signal_activate_flighttrack
-        self.mainwindow_signal_activate_operation = mainwindow.signal_activate_operation
-        self.mainwindow_signal_login_mscolab = mainwindow.signal_login_mscolab
-        self.mainwindow_signal_logout_mscolab = mainwindow.signal_logout_mscolab
         self.mainwindow_listFlightTracks = mainwindow.listFlightTracks
```
Comment (ReimarBauer) [inline mslib/msui/topview.py:204]: those lines were duplicated

File: mslib/msui/topview.py [hunk 2]
```diff
@@ -362,7 +359,7 @@ def openTool(self, index):
                                                             mscolab_server_url=self.mscolab_server_url,
                                                             token=self.token)
 
-                self.mainwindow_signal_logout_mscolab.connect(lambda: self.signal_logout_mscolab.emit())
```
Comment (ReimarBauer) [inline mslib/msui/topview.py:365]: as discussed in https://github.com/Open-MSS/MSS/pull/2310#commits-pushed-0dd6124 this was the culprit

File: mslib/msui/topview.py [hunk 3]
```diff
@@ -362,7 +359,7 @@ def openTool(self, index):
...
+                self.mainwindow_signal_logout_mscolab.connect(self.signal_logout_mscolab.emit)
```
Comment (ReimarBauer) [inline mslib/msui/topview.py:362]: as discussed in #2310 (commits) this was the culprit

Comment (matrss) [reply mslib/msui/topview.py:362]: Could you please file an issue to change this type of lambda everywhere? I've seen it in other places as well and the construct is generally problematic.

File: tests/_test_msui/test_mscolab.py [hunk 4]
```diff
@@ -411,6 +411,74 @@ def assert_logout_text():
...
+            # check 1 view opened
+            assert len(self.window.get_active_views()) == 1
+        qtbot.wait_until(assert_active_views)
+        topview_0 = self.window.listViews.item(0)
+        assert topview_0.window.tv_window_exists is True
+        topview_0.window.setAttribute(QtCore.Qt.WA_DeleteOnClose)
+
+        def assert_attribute():
+            assert topview_0.window.testAttribute(QtCore.Qt.WA_DeleteOnClose)
+        qtbot.wait_until(assert_attribute)
```
Comment (matrss) [inline tests/_test_msui/test_mscolab.py:442]: I am pretty sure that setAttribute is an immediate action that does not require this kind of wait, but I wouldn't know how to find out if this is definitely the case.

File: tests/_test_msui/test_mscolab.py [hunk 5]
```diff
@@ -411,6 +411,74 @@ def assert_logout_text():
...
+            topview_0.window.close()
+
+        def assert_window_closed():
+            assert topview_0.window.tv_window_exists is False
+        qtbot.wait_until(assert_window_closed)
+
+        def assert_label_text():
+            # verify logged in
+            assert self.window.usernameLabel.text() == self.userdata[1]
+        qtbot.wait_until(assert_label_text)
```
Comment (matrss) [inline tests/_test_msui/test_mscolab.py:468]: The `self._login` method is implemented in a way that the login should be finished by the time it returns (i.e. it does such a wait internally). That means this wait_until _could_ be a plain assert, same with where it is used again further down. This only applies to the login though, not to the logout.

**Review by matrss — APPROVED** [review]
LGTM. I left some comments where it could be slightly improved, but nothing critical.

File: mslib/msui/topview.py (hunk 3 above)
Comment (ReimarBauer) [reply mslib/msui/topview.py:362]: a follow up: https://github.com/Open-MSS/MSS/issues/2320

**Pull request merged**